cmake/DAGMC_macros.cmake

```diff
@@ -222,11 +225,7 @@ macro (dagmc_install_library lib_name)
   endif ()
 
   # Keep a list of all libraries being installed
-  if (DAGMC_LIBRARIES)
-    set(DAGMC_LIBRARIES "${DAGMC_LIBRARIES} ${lib_name}" CACHE INTERNAL "DAGMC_LIBRARIES")
-  else ()
-    set(DAGMC_LIBRARIES ${lib_name} CACHE INTERNAL "DAGMC_LIBRARIES")
-  endif ()
+  set(DAGMC_LIBRARIES ${DAGMC_LIBRARIES} ${lib_name} CACHE INTERNAL "DAGMC_LIBRARIES")
```

Comment:
```suggestion
  set(DAGMC_LIBRARIES "${DAGMC_LIBRARIES} ${lib_name}" CACHE INTERNAL "DAGMC_LIBRARIES")
```

Reply:
The current version gives me a semicolon-delimited list. The change above makes it a whitespace deliniated list. 

If I remember right, I've heard complaints about getting this kind of list in a configuration file because they can cause problems when used in different ways externally.  

A: `target_link_libraries(foo_lib ${DAGMC_LIBRARIES}`
B: `target_link_libraries(foo_lib "${DAGMC_LIBRARIES")`

If we write a semicolon-delimited list, case B will fail as the variable won't be expanded but handed verbatim to the command. But if we use a whitespace delineated list, both will be fine. This line of thought changes when working with other entries in a list that might have spaces (like filenames or paths), but for a list of library names I think we're ok. Of course, it isn't a problem if external applications use the variable correctly, but I don't think it hurts us to accommodate here.

That's my understanding anyway and I'm happy to defer to you on it!

Reply:
Any thoughts on this either way @ljacobson64?

Reply:
I guess my experience is that I've had a lot more trouble dealing with lists formatted as strings than I have with lists formatted as lists

Reply:
Good enough for me!